### PR TITLE
pkgconf: remove compiler from package_id

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **pkgconf/***

fixes conan-io/conan-center-index#7540

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by uilianries — APPROVED** [review]
LGTM

https://github.com/conan-io/conan-center-index/blob/154c9edca773d52a790d917bf2c0b95796a7ce7b/docs/packaging_policy.md#settings

File: recipes/pkgconf/all/conanfile.py [hunk 1]
```diff
@@ -114,6 +114,9 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, "share"))
         tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
 
+    def package_id(self):
+        del self.info.settings.compiler
```
Comment (madebr) [inline recipes/pkgconf/all/conanfile.py:118]: This recipe also provides a library, which requires a compiler present in the settings + package id.
Unless no other recipe depends on the library, I'm fine with stopping providing it, or optionally providing it as an option.

Comment (ericLemanissier) [reply recipes/pkgconf/all/conanfile.py:118]: owww, I forgot that right. I'll add an option.

Comment (uilianries) [reply recipes/pkgconf/all/conanfile.py]: @madebr good catch!

**Review by uilianries — APPROVED** [review]

**Review by madebr — COMMENTED** [review]
I think the autoconf integration should still work with enable_lib disabled.

**Review by uilianries — APPROVED** [review]

**Comment by ericLemanissier** [comment]
This is ready for review, and the CI passed, contrary to what the checks statuses report

**Review by SSE4 — APPROVED** [review]

**Pull request merged**